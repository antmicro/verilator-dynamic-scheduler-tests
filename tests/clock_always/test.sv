module t;
   /* verilator lint_off UNOPTFLAT */
   logic clk = 0;
   int   cyc = 0;

   always #1 clk = ~clk;

   always @(negedge clk) begin
      $write("[%0t] negedge; clk == %b\n", $time, clk);
   end

   always @(posedge clk) begin
      cyc <= cyc + 1;
      $write("[%0t] posedge; clk == %b\n", $time, clk);
      if (cyc == 10) begin
         $write("*-* All Finished *-*\n");
         $finish;
      end
   end

endmodule
