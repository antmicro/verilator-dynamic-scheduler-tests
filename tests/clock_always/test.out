[1] posedge; clk == 1
[2] negedge; clk == 0
[3] posedge; clk == 1
[4] negedge; clk == 0
[5] posedge; clk == 1
[6] negedge; clk == 0
[7] posedge; clk == 1
[8] negedge; clk == 0
[9] posedge; clk == 1
[10] negedge; clk == 0
[11] posedge; clk == 1
[12] negedge; clk == 0
[13] posedge; clk == 1
[14] negedge; clk == 0
[15] posedge; clk == 1
[16] negedge; clk == 0
[17] posedge; clk == 1
[18] negedge; clk == 0
[19] posedge; clk == 1
[20] negedge; clk == 0
[21] posedge; clk == 1
*-* All Finished *-*
